modules/openapi-generator/src/main/resources/dart2/serialization/native/native_class.mustache

```diff
@@ -75,10 +75,10 @@ class {{{classname}}} {
   static {{{classname}}} fromJson(Map<String, dynamic> json) => {{{classname}}}(
   {{#vars}}
     {{#isDateTime}}
-        {{{name}}}: mapDateTime(json, r'{{{baseName}}}', '{{{pattern}}}'),
+        {{{name}}}: mapDateTime(json, r'{{{baseName}}}', '{{{pattern}}}'){{^isNullable}}!{{/isNullable}},
```

Comment:
shouldn't you check for `isRequired` as well ?

Reply:
as in `if (required and notNullable) OR (notRequired and nullable)`?

Reply:
well, that's what I was discussing in https://github.com/OpenAPITools/openapi-generator/issues/10972 
if the date can be `notNullable` and `notRequired`, should we assume it's nullable or not?

Reply:
I saw that discussion, but I dont have a standpoint there. If we (or you ;)) decide on a route to go, we should migrate all dart generators to make them compatible with each other. Until this is the case, I'm happy to go whichever you guys decide as a temporary solution. only nullable checks is my way as it's "dart-null" and "openapi-null", it is the same semantic, despite it has different meanings.

Reply:
Im leaving this as is, but here's a snippet for future if needed `{{#required}}{{^isNullable}}!{{/isNullable}}{{/required}}{{^required}}{{#isNullable}}!{{/isNullable}}{{/required}}`

Reply:
I used the following in my pr, imho it's a better one because it takes into consideration the `defaultValue`:

`{{#required}}{{^isNullable}}!{{/isNullable}}{{/required}}{{^required}}{{#defaultValue}} ?? {{{.}}}{{/defaultValue}}{{/required}}`

The pr: https://github.com/OpenAPITools/openapi-generator/pull/11149